src/swap/SwapScreen.tsx

```diff
@@ -220,6 +224,67 @@ function getNetworkFee(quote: QuoteResult | null, networkId?: NetworkId) {
   }
 }
 
+function useCrossChainFee(
+  quote: QuoteResult | null,
+  sourceNetworkId: NetworkId
+): {
+  crossChainFeeToken: TokenBalance
+  maxCrossChainFeeAmount: BigNumber
+  maxCrossChainFeeMissingAmountInDecimal: BigNumber
+} {
+  // Fee currencies arrays always have the native currency first, and cross chain
+  // swap fees are always paid with the chain's native currency.
+  const feeCurrency = useSelector((state) => feeCurrenciesSelector(state, sourceNetworkId))[0]
+
+  if (quote?.unvalidatedSwapTransaction.type !== 'cross-chain') {
+    return {
+      crossChainFeeToken: feeCurrency,
+      maxCrossChainFeeAmount: new BigNumber(0),
+      maxCrossChainFeeMissingAmountInDecimal: new BigNumber(0),
+    }
+  }
+
+  let gasNeededInCrossChainFeeCurrency = new BigNumber(0)
+
+  // Gas is going to eat into our budget for the cross chain fee
+  if (
+    quote.preparedTransactions.type === 'possible' &&
+    quote.preparedTransactions.feeCurrency.tokenId === feeCurrency.tokenId
+  ) {
+    gasNeededInCrossChainFeeCurrency = getEstimatedGasFee(quote.preparedTransactions.transactions)
```

Comment:
hmm, yeah.. added `maxGasFeeInDecimal` to `PreparedTransactionsPossible` (it doesn't make sense to add it for `PreparedTransactionsNotEnoughBalanceForGas` however)

Reply:
Actually, the immediate code change is simple, but it requires a lot of cascading changes elsewhere (many tests) so I'm opting against it for this PR since it's already pretty big.